### PR TITLE
fix panic when VolumeScheduling=False

### LINKED ISSUE #70146: scheduler 1.12.1 sigsegv with nil pointer
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug
/sig scheduling

**What happened**:

```
I1023 19:19:05.400206       1 scheduler.go:298] Trying to bind volumes for pod "kube-system/kube-flannel-8dcq5"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x36b1455]

goroutine 294 [running]:
k8s.io/kubernetes/pkg/scheduler.(*Scheduler).bindVolumes(0xc42000fc30, 0xc4209cea80, 0xc42023a810, 0xc42023a818)
	/workspace/anago-v1.12.1-beta.0.52+4ed3216f3ec431/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/scheduler.go:299 +0x135
k8s.io/kubernetes/pkg/scheduler.(*Scheduler).scheduleOne.func1(0xc4204beb00, 0xc42000fc30, 0xc4209cea80, 0xc4208ea6c0, 0xc4204beb10, 0x6, 0xbeebfa7251a9dac1, 0x41952c035, 0x956d500)
	/workspace/anago-v1.12.1-beta.0.52+4ed3216f3ec431/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/scheduler.go:448 +0x26a
created by k8s.io/kubernetes/pkg/scheduler.(*Scheduler).scheduleOne
	/workspace/anago-v1.12.1-beta.0.52+4ed3216f3ec431/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/scheduler.go:445 +0x7a6
```

**What you expected to happen**:

If the scheduler was expecting something to happen that did not, it should emit a log message rather than a nil pointer that doesn't help anyone know any kind of corrective action to take

**How to reproduce it (as minimally and precisely as possible)**:

```yaml
  containers:
  - name: kube-scheduler
    image: gcr.io/google-containers/hyperkube-amd64:v1.12.1
    imagePullPolicy: IfNotPresent
    resources:
      limits:
        cpu: 250m
        memory: 512M
      requests:
        cpu: 80m
        memory: 170M
    command:
    - /hyperkube
    - scheduler
    - --leader-elect=true
    - --kubeconfig=/etc/kubernetes/kube-scheduler-kubeconfig.yaml
    - --profiling=False
    - --v=2
    - --feature-gates=PersistentLocalVolumes=False,VolumeScheduling=False,MountPropagation=False
```

**Anything else we need to know?**:

It appears that removing `VolumeScheduling=False,` from the `feature-gates` cures the nil pointer

**Environment**:
- Kubernetes version (use `kubectl version`): Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.1", GitCommit:"4ed3216f3ec431b140b1d899130a69fc671678f4", GitTreeState:"clean", BuildDate:"2018-10-05T16:36:14Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
- Cloud provider or hardware configuration: virtualbox
- OS (e.g. from /etc/os-release): Ubuntu 18.04.1 LTS
- Kernel (e.g. `uname -a`): Linux k8s-02 4.15.0-36-generic #39-Ubuntu SMP Mon Sep 24 16:19:09 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: kubespray v2.7.0
- Others: docker 18.06.1-ce

### PR DESCRIPTION
**Pull request by Pingan2017**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #70146 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix scheduler's panic when set `--feature-gates=VolumeScheduling=False`
```

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -546,7 +546,7 @@ func (sched *Scheduler) scheduleOne() {
 	// bind the pod to its host asynchronously (we can do this b/c of the assumption step above).
 	go func() {
 		// Bind volumes first before Pod
-		if !allBound {
+		if !allBound && utilfeature.DefaultFeatureGate.Enabled(features.VolumeScheduling) {
```
Comment (wgliang) [inline pkg/scheduler/scheduler.go]: Maybe handle it in assumeVolumes phase will be better.

Comment (Pingan2017) [reply pkg/scheduler/scheduler.go]: done. PTAL

File: pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -367,8 +367,9 @@ func (sched *Scheduler) assumeVolumes(assumed *v1.Pod, host string) (allBound bo
 			invalidPredicates := sets.NewString(predicates.CheckVolumeBindingPred)
 			sched.config.Ecache.InvalidatePredicates(invalidPredicates)
 		}
+		return allBound, err
 	}
-	return
+	return true, nil
```
Comment (neolit123) [inline pkg/scheduler/scheduler.go]: this seems odd to me.
the original code looks fine, unless there is a catch here.

**Comment by neolit123** [comment]
thanks for the attempt to fix this @Pingan2017 
please add a release note instead of `NONE` that explains the change in one sentence.

File: pkg/scheduler/scheduler.go (hunk 2 above)
Comment (Pingan2017) [reply pkg/scheduler/scheduler.go]: when set `--feature-gates=VolumeScheduling=False` , `assumeVolumes` should return true to skip func `bindVolumes`. now it retrun false(default), cause `bindVolumes` panic
#70146

**Comment by Pingan2017** [comment]
@wgliang @k82cn @neolit123 PTAL , Thank u

**Comment by Pingan2017** [comment]
cc @k82cn Can u take a look ?

**Comment by k82cn** [comment]
> cc @k82cn Can u take a look ?

Sure, let me check it this weekend :)

**Comment by k82cn** [comment]
please add/update an/the e2e to catch that.

**Comment by Pingan2017** [comment]
@k82cn OK, I'll add/update it.  but I have not work on e2e test before.  Is there some example for scheduler?
BTW, Should I file an new PR for e2e test OR add a commit in this PR?

**Comment by wgliang** [comment]
@Pingan2017 Will you continue to do this work?

**Pull request closed without merging**